Clarify project.yaml usage (#1796)
puppetlabs/bolt: documentation/directory_structure.md

-   As a best practice, write automated tests for the tasks and plans in your module, if possible. For information about automated testing patterns, check out these resources: [Example of unit testing plans and integration \(acceptance\) testing tasks](https://github.com/puppetlabs/puppetlabs-facts) (GitHub) and [Writing Robust Puppet Bolt Tasks: A Guide](https://puppet.com/blog/writing-robust-puppet-bolt-tasks-guide) (Puppet blog).

## Simplified directory structure

If you're working on a new project and you want to use a simplified directory
structure, you can load your plan and task content directly from your project as
if it was a module. 

> **Note:** This is an experimental feature.

To load your content from a Bolt project:
1. Create a `project.yaml` file in the root directory of your Bolt project. 
2. Place your Bolt plans and tasks in `plans` and `tasks` directories
   respectively.

> If `project.yaml` exists at the root of a project directory, and your tasks and
> plans are located in their respective directories, Bolt loads the project as a 
> module namespaced to the name of the directory.

Reviewer: ```suggestion
If `project.yaml` exists at the root of a project directory, Bolt loads the project as a module.
Tasks and plans are loaded from the `tasks/` and `plans/` directories and are namespaced
to the project name.
```

You don't need any tasks or plans for Bolt to load the project as a module. If the directories are missing or don't have any content, the project is loaded like any other module that doesn't have tasks or plans.

Here is an example of a project using a simplified directory structure:
```console
.
├── bolt.yaml
├── plans
│   └── myplan.yaml
├── project.yaml
└── tasks

>     └── mytask.yaml

Reviewer: Tasks won't be a yaml file -- another extension like `.rb` or `.sh` would work better here.
